quinn-proto/src/connection/streams/mod.rs

```diff
@@ -233,6 +233,11 @@ impl<'a> SendStream<'a> {
         self.write_source(&mut BytesArray::from_chunks(data))
     }
 
+    /// Returns the maximum amount of data this is allowed to be written on the connection.
+    pub fn write_limit(&self) -> u64 {
+        self.state.write_limit()
```

Comment:
This is the limit for the lifetime number of bytes sent on the entire connection. You also need to account for `self.max_data`.